### ISSUE TITLE
S2583 FP: Condition evaluates to constant - false positive when casting floating point numbers

### ISSUE THREAD

**Issue by valhristov** [SonarSource/sonar-dotnet#739]
### Description

False positive is detected in the following code:
```
var x = 5.5;
var y = (int)x;
if (x == y) // always true
```

Note: similar FP with the same core reason could be detected in other rules as well.

[original SO question](https://stackoverflow.com/questions/45996751/change-this-condition-so-that-it-does-not-always-evaluate-to-true)

**Comment by mary-georgiou-sonarsource** [comment]
Fixed in https://github.com/SonarSource/sonar-dotnet/pull/7750
